src/persist-client/src/write.rs

```diff
@@ -449,6 +449,8 @@ where
                     desc: desc.clone(),
                     keys,
                     len: num_updates,
+                    // WIP: should we compute runs while appending batches?
```

Comment:
thinking about this more. what happens if we don't? we end up with a run that isn't sorted and the first compaction with that batch fixes it? besides that being surprising if we were to have a metric for it (side note: 🤔), seems pretty okay?

Reply:
The first compaction wouldn't necessarily fix it if it's more than 1 part, I think if the batch were wholly unsorted and split into N parts, the first compaction would turn it into N sorted parts, each of which is potentially its own run. Further compaction afterwards would help consolidate those runs together

I've been part wondering if we want to move any write-side run tracking into a future PR, since correctness-wise compaction should still work without it

Reply:
okay so I started to poke at this one and it quickly becomes hairy / likely worthy of its own PR if we do pursue this. currently we feed updates straight into the columnar format without sorting, so there's not an easy way to track the greatest update / runs along the way. if we wanted to do so, I think we'd need to restructure the way Batch chunks up updates similarly to what's been done here for compaction